Separate is_cuda() and is_sparse() in TORCH_CHECK

mjanderson09 opened the pull request:
Summary:
Switch checks in `check_gpu_single_tensor(...)`
From:
   if(!stmt) -> TORCH_CHECK(false, mst)
 to
   TORCH_CHECK(stmt, msg)

Also separated out the `is_cuda()` check from the `is_sparse()` check so that the message is more clear.

Test Plan: ci

Differential Revision: D45103310

kumpera reviewed (approved):
That's a neat improvement to error reporting. Thanks for the contribution.

mjanderson09 committed:
Separate is_cuda() and is_sparse() in TORCH_CHECK (#99641)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/99641

Switch checks in `check_gpu_single_tensor(...)`
From:
   if(!stmt) -> TORCH_CHECK(false, mst)
 to
   TORCH_CHECK(stmt, msg)

Also separated out the `is_cuda()` check from the `is_sparse()` check so that the message is more clear.

Test Plan: ci

Differential Revision: D45103310

fbshipit-source-id: 2c7c622c36940011795d55aeb8cc93af00da2b17
